charts/thanos/templates/query-frontend/hpa.yaml

```
@@ -1,4 +1,4 @@
-{{- if and (and .Values.queryFrontend.enabled .Values.queryFrontend.autoscaling.enabled) (or .Values.queryFrontend.autoscaling.targetCPUUtilizationPercentage .Values.queryFrontend.autoscaling.targetMemoryUtilizationPercentage) -}}
+{{- if and .Values.query.enabled (and .Values.queryFrontend.enabled .Values.queryFrontend.autoscaling.enabled) (or .Values.queryFrontend.autoscaling.targetCPUUtilizationPercentage .Values.queryFrontend.autoscaling.targetMemoryUtilizationPercentage) -}}
```

```suggestion
{{- if and .Values.query.enabled (and (and .Values.queryFrontend.enabled .Values.queryFrontend.autoscaling.enabled) (or .Values.queryFrontend.autoscaling.targetCPUUtilizationPercentage .Values.queryFrontend.autoscaling.targetMemoryUtilizationPercentage)) -}}
```

Query disabled should logically short circuit other checks.